app/scripts/lib/Stores/ExtensionStore.ts

```diff
@@ -0,0 +1,70 @@
+import browser from 'webextension-polyfill';
+import log from 'loglevel';
+import { checkForLastError } from '../../../../shared/modules/browser-runtime.utils';
+import {
+  type IntermediaryStateType,
+  MetaMaskStorageStructure,
+  BaseStore,
+} from './BaseStore';
+
+/**
+ * An implementation of the MetaMask Extension BaseStore system that uses the
+ * browser.storage.local API to persist and retrieve state.
+ */
+export default class ExtensionStore extends BaseStore {
+  isSupported: boolean;
+
+  constructor() {
+    super();
+    this.isSupported = Boolean(browser.storage.local);
+    if (!this.isSupported) {
+      log.error('Storage local API not available.');
+    }
+  }
+
+  /**
+   * Returns all of the keys currently saved
+   *
+   * @returns the key-value map from local storage
```

Comment:
Nit: Maybe misleading to say this returns "keys", when it returns all data. Also better to avoid calling this "local storage", to avoid confusion with the other web API with that name.

```suggestion
   * Return all data in `local` extension storage area.
   *
   * @returns All data stored`local` extension storage area.
```